src/audio/volume.c

```diff
@@ -448,7 +448,9 @@ static int volume_copy(struct comp_dev *dev)
 	cd->scale_vol(dev, sink, source);
 
 	/* calc new free and available */
+	dcache_writeback_region(sink->w_ptr, cd->sink_period_bytes);
```

Comment:
Maybe it would be better to add dcache_writeback_region inside comp_update_buffer_consume and dcache_invalidate_region inside comp_update_buffer_produce method.